neurox/data/extraction/transformers_extractor.py

```diff
@@ -173,7 +176,7 @@ def extract_sentence_representations(
                 hidden_states[0].cpu().numpy()
                 for hidden_states in all_hidden_states[1:]
             ]
-        all_hidden_states = np.array(all_hidden_states)
+        all_hidden_states = np.array(all_hidden_states).astype(dtype)
```

Comment:
Slightly more efficient to specify dtype as an argument to np.array (so `np.array(all_hidden_states, dtype=dtype)`